### PR TITLE
[Feature] Add login method for third party authorisation

### PR DESCRIPTION
**Pull request by maltsavkiryl**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -86,8 +85,23 @@ export class OAuth2VueClient {
     this.client = null
   }
 
-  public async login(username: string, password: string): Promise<void> {
-    const client = await this.oAuthFactory.login(username, password)
+  public async loginPassword(username: string, password: string): Promise<void> {
+    if (this.options.isMock)
```
Comment (JeroenVanC) [inline src/index.ts:89]: if'ke zonder brackets :)

File: src/index.ts [hunk 2]
```diff
@@ -86,8 +85,23 @@ export class OAuth2VueClient {
...
+    const client = await this.oAuthFactory.loginPassword(username, password)
+
+    const tokens = client.getTokens()
+
+    this.saveTokensToLocalStorage(tokens)
+    this.client = this.createClient(tokens)
+  }
+
+  public async loginAuthorisation(code: string, state: string, grantType: OAuth2ClientGrantType): Promise<void> {
+    if (this.options.isMock)
```
Comment (JeroenVanC) [inline src/index.ts:101]: ifke zonder brackets

**Review by JeroenVanC — APPROVED** [review]
2 if'kes zonder brackets

File: src/index.ts (hunk 1 above)
Comment (maltsavkiryl) [reply src/index.ts:89]: da is die stomme linter die dat automatisch weghaalt

**Review by wouterlms — APPROVED** [review]

**Pull request merged**